### ISSUE TITLE
CI/CD: Add label automation to contrib repo's discussions

### ISSUE THREAD

**Issue by crobert-1** [open-telemetry/opentelemetry-collector-contrib#27629]
### Describe the issue you're reporting

This is being created from discussions around #27456. A lot of issues are being created that would be better suited for discussions, as they're often usability questions. In order to better support discussions we need to be able to add labels to discussions that can ping code owners. Once this is in place we can add a new issue template that will automatically convert questions to discussions.

**Comment by crobert-1** [comment]
I've been able to take a bit of a look at this and see the status/blockers. We can currently make use of some of the same scripting that we use for issues, such as the `add_labels.yml` [workflow](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/add-labels.yml), since there are the same [actions in discussions](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#discussion) that can trigger workflows.

However, it looks like the current problem is that the gh CLI (and API) does not interface with discussions, which is how we [add labels to issues](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/784514fb0440f8cb29e36671e6ed1854c41b12fc/.github/workflows/scripts/add-labels.sh#L41).

The current solution provided by GitHub is to use GraphQL for discussion API. Sources: [here](https://github.com/orgs/community/discussions/4327), and [here.](https://github.com/orgs/community/discussions/23483)

[Documentation reference.](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions)

Example queries for adding labels to a discussion: [Here](https://github.com/orgs/community/discussions/15431), and [here.](https://github.com/cli/cli/discussions/7973)

**Comment by mowies** [comment]
A nice solution for this could be the [`addLabelsToLabelable`](https://docs.github.com/en/graphql/reference/mutations#addlabelstolabelable) and [`removeLabelsFromLabelable`](https://docs.github.com/en/graphql/reference/mutations#removelabelsfromlabelable) GraphQL mutations.
Since both issues and discussions are [labelable](https://docs.github.com/en/graphql/reference/interfaces#labelable), we could abstract away the difference that way.

**Comment by niharikag09** [comment]
Hi, I'd be happy to take this up. Please let me know if this issue is still desired and available. Will cover the following:

• Proposing extending  .github/workflows/add-labels.yml  for discussion comments.
• Reusing  .github/workflows/scripts/add-labels.sh  parsing/authorization logic.
• Using GraphQL  addLabelsToLabelable  and  removeLabelsFromLabelable .
• Adding tests for add/remove, invalid labels, and unauthorized users.

Also let me know if you prefer adding a new script or using the existing script.
